pkg/controllers/disruption/eventual.go

```diff
@@ -86,7 +88,7 @@ func (d *Drift) ComputeCommand(ctx context.Context, disruptionBudgetMapping map[
 		// If the disruption budget doesn't allow this candidate to be disrupted,
 		// continue to the next candidate. We don't need to decrement any budget
 		// counter since drift commands can only have one candidate.
-		if disruptionBudgetMapping[candidate.nodePool.Name][d.Reason()] == 0 {
+		if disruptionBudgetMapping[candidate.nodePool.Name] == 0 {
```

Comment:
I know we mark if something is constrained by budgets in other implementations -- is there a reason that we don't do the same thing here?

Reply:
So if I understand it correctly `constrainedByBudgets` is used to track if karpenter will consider the cluster as consolidated. https://github.com/engedaam/karpenter-core/blob/1595ce081e448e40f97ea84fa649e84f5ae655ac/pkg/controllers/disruption/multinodeconsolidation.go#L92 For eventual disruption, I don't think the idea make sense.

Reply:
Yep, makes sense. You're right that we don't need this.